src/errors.ts

```diff
@@ -0,0 +1,77 @@
+import axios from "axios";
+import * as vscode from "vscode";
+import { stopEnvironmentPolling } from "./extension";
+
+const reLoginButtonText = "Re-Login";
+const onReLoginClicked = async (buttonText?: string) => {
+  if (buttonText === reLoginButtonText) {
+    await vscode.commands.executeCommand("env0.logout");
+    await vscode.commands.executeCommand("env0.login");
+  }
+};
+
+const onUnauthorized = async (errorMessage: string) => {
+  const buttonClicked = await vscode.window.showErrorMessage(
+    `${errorMessage}: unauthorized, please re-login`,
+    reLoginButtonText
+  );
+  await onReLoginClicked(buttonClicked);
+};
+
+const onForbidden = async (errorMessage: string) => {
+  const buttonClicked = await vscode.window.showErrorMessage(
+    `${errorMessage}: forbidden, please check your credentials`,
+    reLoginButtonText
+  );
+  await onReLoginClicked(buttonClicked);
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
please let's avoid those
`// eslint-disable-next-line @typescript-eslint/no-explicit-any`
comments...  isn't it `Error` type ?

Reply:
In theory in JS you can `throw` everything that you want, but for the current implementation it doesn't matter, so i changed it to Error 
114f63390053085f1179bcf6a6298d540002e482